script/buildSQLGeom.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+
+"""
+This module will accept an input text file with (x,y,[z]) coordinates in a CSV-format list
+and generate a SQL string that will add these coordinates as a PostGIS Polygon
+to sites.geometry within a BETYdb instance.
+
+A header row should be included in the input file, although the column names do not matter. If a z column
+for altitude is not included, the script will attempt to get it from USGS NED (US) or Google (global).
```

Comment:
USGS is preferred because it is more accurate and consistent. The accuracy
of Google results is inversely proportional to the number of points.

Id go with 1. Query USGS. 2. If no data query Google.
On Thu, Nov 19, 2015 at 4:45 PM Max Burnette <notifications@github.com>
wrote:

> In script/buildSQLGeom.py
> <https://github.com/PecanProject/bety/pull/369#discussion_r45412745>:
>
> > @@ -0,0 +1,236 @@
> > +#!/usr/bin/env python
> > +
> > +"""
> > +This module will accept an input text file with (x,y,[z]) coordinates in a CSV-format list
> > +and generate a SQL string that will add these coordinates as a PostGIS Polygon
> > +to sites.geometry within a BETYdb instance.
> > +
> > +A header row should be included in the input file, although the column names do not matter. If a z column
> > +for altitude is not included, the script will attempt to get it from USGS NED (US) or Google (global).
>
> In the current code, *everything* will go through USGS. My thinking was
> that if you have a valid key, we could just make everything go through
> Google. Do you consider USGS the preferred authority in the US?
>
> We can make USGS domestic default and Google international default, but we
> need a good way to determine whether the point is in the US to begin with,
> which is likely another API call. Perhaps the anticipated traffic is low
> enough that this is fine.
>
> Another option would be to query USGS, and if that doesn't find anything,
> we query Google. Or, we could define a simple rectangle around the US and
> quickly check whether the lat/lon exceed that boundary, but that falls
> apart e.g. in northern Mexico. Don't love that idea.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/PecanProject/bety/pull/369/files#r45412745>.
>